symengine/ruby/ext/symengine/ruby_basic.c

```diff
@@ -72,6 +76,90 @@ VALUE cbasic_pow(VALUE self, VALUE operand2){
     return cbasic_binary_op(self, operand2, basic_pow);
 }
 
+VALUE cbasic_eq(VALUE self, VALUE operand2) {
+    basic_struct *this, *cbasic_operand2;
+
+    Data_Get_Struct(self, basic_struct, this);
+    Data_Get_Struct(operand2, basic_struct, cbasic_operand2);
+
+    return basic_eq(this, cbasic_operand2) ? Qtrue : Qfalse;
+}
+
+VALUE cbasic_neq(VALUE self, VALUE operand2) {
+    basic_struct *this, *cbasic_operand2;
+
+    Data_Get_Struct(self, basic_struct, this);
+    Data_Get_Struct(operand2, basic_struct, cbasic_operand2);
+
+    return basic_neq(this, cbasic_operand2) ? Qtrue : Qfalse;
+}
+
 VALUE cbasic_neg(VALUE self){
     return cbasic_unary_op(self, basic_neg);
 }
+
+VALUE cbasic_get_args(VALUE self) {
+    basic_struct *this, *iterator_basic;
+    CVecBasic *args = vecbasic_new();
+    int size = 0;
+
+    Data_Get_Struct(self, basic_struct, this);
+    basic_get_args(this, args);
+
+    size = vecbasic_size(args);
+    VALUE ruby_array = rb_ary_new2(size);
+    int i = 0;
+    VALUE temp = NULL;
+    for(i = 0; i < size; i++) {
+        basic_init(iterator_basic);
+        vecbasic_get(args, i, iterator_basic);
+        temp = Data_Make_Struct(rb_obj_class(self), basic_struct, NULL,
+                                                    cbasic_free, iterator_basic);
+        rb_ary_push(ruby_array, temp);
+        basic_free(iterator_basic);
```

Comment:
Much better. Then there is another problem. `basic_init` initializes *inplace*. So this will cause a segfault. We need to extend our C interface to allow both static and dynamic allocation.